src/components/app/ProfileFilterNavigator.js

```diff
@@ -21,10 +21,12 @@ type StateProps = $ReadOnly<$Exact<$Diff<Props, DispatchProps>>>;
 const options: ExplicitConnectOptions<{||}, StateProps, DispatchProps> = {
   mapStateToProps: state => {
     const items = getRangeFilterLabels(state);
+    const profileSelection = state.profileView.viewOptions.selection;
```

Comment:
To get the selection, we usually use selectors. We do have one here: https://github.com/devtools-html/perf.html/blob/54f197d7a100cedfb9a9f3d0d7136648d92ed6da/src/reducers/profile-view.js#L526-L527

Then I think you can calculate the selection duration here, and pass it directly as an "uncommitedItem" as was suggested by Greg in #422, so that `FilterNavigatorBar` stays generic.